api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -765,6 +765,21 @@ message HttpConnectionManager {
   // If this config is set, the Proxy-Status HTTP response header field is
   // populated. By default, it is not.
   ProxyStatusConfig proxy_status_config = 49;
+
+  // Configuration options for Header Validation (UHV).
```

Comment:
How does this interact with existing fields like `path_with_escaped_slashes_action` and `path_normalization_options`?

Reply:
I have updated comments in the http_connection_manager. The new config overrides existing options in the http_connection_manager. I will be updating it as we move more of the existing options into this extension.

Reply:
and maybe reject config if they're both set?

Reply:
I am fine with logging a warning.

> Setting field C in "new config" and having it invalidate A and B leads to outages.

I certainly accept that that's possible, because people make mistakes all the time, but it's not necessarily clear to me that this is significantly more likely with this behavior than it would be with any field in the API.  If the semantics of the new field are spelled out to override the old fields, then it shouldn't be a surprise to anyone, and people really need to understand the semantics of any field before they set it.

> Theoretically the whole design of the long lived old config allows for the same for Envoy APIs and slower-migrating clients.

I think we need to be sensitive to control planes that support both old and new clients.  When you take into account the need for xDS resources to be cacheable, it actually becomes fairly complicated for control planes to generate different variants of the same resource for different clients.  I think it makes life much easier for control planes when they can provide a single resource that works for both old and new clients.

Reply:
I'll add a TODO for now and change it once the fields are marked as deprecated.